### PR TITLE
Release 1.8.0-rc.3

### PR DESCRIPTION
**Pull request by Jack-Works**
After the basic test, 1.8.0-rc.1 can be release on the related platform.

# Chrome Web Store
- [x] Install as a new user, setup Facebook
- [x] Install as a new user, setup Twitter
- [x] Install as a new user, setup Facebook and Twitter
- [x] Setuped in 1.7.1, then upgrade to 1.8.0-rc.1
- [x] Install as a return user
- [x] Release 1.8.0-rc.1 on the Chrome Web Store

# Firefox Addon
- [x] Install as a new user, setup Facebook
- [ ] Install as a new user, setup Twitter
- [ ] Install as a new user, setup Facebook and Twitter
- [ ] Install as a return user
- [ ] Setuped in 1.7.1, then upgrade to 1.8.0-rc.1
- [ ] (Firefox for Android) Install as a new user, setup Facebook
- [ ] (Firefox for Android) Install as a new user, setup Twitter
- [ ] (Firefox for Android) Install as a new user, setup Facebook and Twitter
- [ ] (Firefox for Android) Install as a return user
- [ ] (Firefox for Android) Setuped in 1.7.1, then upgrade to 1.8.0-rc.1
- [x] Release 1.8.0-rc.1 on the Firefox Addon

# Release on GitHub
- [ ] changelog
    - [ ] Release on Github

# Changes:
https://github.com/DimensionDev/Maskbook/compare/v1.7.1...401ed24318622627103f42376fbf6046d7ed96fc

### CODE REVIEW

**Comment by Jack-Works** [comment]
Problems:
- [ ] DOMException: Version change transaction was aborted in upgradeneeded event handler.

**Comment by Jack-Works** [comment]
Okay, don't merge it. We'll leave it open for a window to fix bugs in the next week.

File: src/utils/comparer.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Group, Person } from '../database'
+
+export const PersonArrayComparer = (a: Person[], b: Person[]) => {
+    if (a.length !== b.length) return false
+    return a.every((person, index) => {
+        const target = b[index]
```
Comment (Jack-Works) [inline src/utils/comparer.ts]: Do we need a position-to-position compare?

Comment (SunriseFox) [reply src/utils/comparer.ts]: don't know... You may remove some if you find it unnecessary.

Comment (Artoria2e5) [reply src/utils/comparer.ts]: Is there something like a deepequals library for this stuff?

Comment (Jack-Works) [reply src/utils/comparer.ts]: https://github.com/tc39/proposal-record-tuple is nice to us

File: src/social-network-provider/twitter.com/ui/index.ts [hunk 2]
```diff
@@ -41,4 +44,49 @@ export const instanceOfTwitterUI = defineSocialNetworkUI({
...
+                    return `rgb(${getBackgroundColor().join(',')})`
+                },
+            },
+        },
+    }),
 })
+
+function isDarkMode(): 'dark' | 'light' {
+    const [r, g, b] = getBackgroundColor()
+    if (r < 68 && g < 68 && b < 68) return 'dark'
```
Comment (Artoria2e5) [inline src/social-network-provider/twitter.com/ui/index.ts]: This is some crazy color-theory thing going on here.

Comment (Jack-Works) [reply src/social-network-provider/twitter.com/ui/index.ts]: Neruthus tells me this magic value

Comment (neruthes) [reply src/social-network-provider/twitter.com/ui/index.ts]: Magic number indeed.

**Pull request merged**